Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.cpp

```diff
@@ -0,0 +1,178 @@
+#include <stdlib.h>
+#include <crtdbg.h>
+#include <iostream>
+#include <string>
+
+#include "SLList.h"
+
+template<typename T>
+void PrintList(TSLList<T>& List)
+{
+	std::cout << "Size: " << List.GetSize() << "\t\tList Contents: { ";
+	for (const T& Element : List)
+	{
+		std::cout << Element << ' ';
+	}
+	std::cout << "}\n";
+}
+
+template<typename T>
+void PrintHead(const TSLList<T>& List, const std::string& CorrectHead)
+{
+	std::cout << "Head is:\t" << List.GetHead() << (List.GetHead() == CorrectHead ? " CORRECT" : " INCORRECT") << std::endl;
+}
+
+template<typename T>
+void PrintTail(const TSLList<T>& List, const std::string& CorrectTail)
+{
+	std::cout << "Tail is:\t" << List.GetTail() << (List.GetTail() == CorrectTail ? " CORRECT" : " INCORRECT") << std::endl;
+}
+
+int main()
+{
+	// Memory Leak Detection
+	_CrtSetDbgFlag(_CRTDBG_ALLOC_MEM_DF | _CRTDBG_LEAK_CHECK_DF);
+
+	// Uncomment the next line if you need to check where a memory leak occurred!
+	//_CrtSetBreakAlloc(Code); // <--- replace 'Code' with whichever code you saw in the output window
+
+	// Linked List
+	std::cout << "*******************************************************************************\n";
+	std::cout << "**                           Linked List Test!                               **\n";
+	std::cout << "*******************************************************************************\n\n";
+
+	// Definitions
+	TSLList<std::string> List;
+
+	// Test 1
+	std::cout << "\n*** Test 1 ***\n";
+
+	PrintList(List);
+
+	List.AddHead("Correctly");
+	PrintList(List);
+	List.AddHead("Read");
+	PrintList(List);
+	List.AddHead("Should");
+	PrintList(List);
+	List.AddHead("This");
+	PrintList(List);
+
+	PrintHead(List, "This");
+	PrintTail(List, "Correctly");
+
+	std::cout << std::endl;
+
+	List.Clear();
+	PrintList(List);
+
+	List.AddTail("This");
+	PrintList(List);
+	List.AddTail("Should");
+	PrintList(List);
+	List.AddTail("Read");
+	PrintList(List);
+	List.AddTail("Correctly");
+	PrintList(List);
+
+	PrintHead(List, "This");
+	PrintTail(List, "Correctly");
+
+	std::cout << std::endl;
+
+	List.Clear();
+	List.AddHead("Head");
+	PrintList(List);
+
+	PrintHead(List, "Head");
+	PrintTail(List, "Head");
+
+	std::cout << std::endl;
+
+	List.Clear();
+	List.AddTail("Tail");
+	PrintList(List);
+
+	PrintHead(List, "Tail");
+	PrintTail(List, "Tail");
+	List.Clear();
+
+	// Test 2
+	std::cout << "\n*** Test 2 ***\n";
+	std::cout << "Inserting and removing...";
+
+	List.AddTail("This");
+	List.AddTail("Is");
+	List.AddTail("A");
+	List.AddTail("List");
+
+	std::cout << std::endl;
+	PrintList(List);
+
+	std::cout << "\nRemoving Is...\n";
+	List.Remove(1);
+	PrintList(List);
+
+	std::cout << "\nRemoving invalid indices... shouldn't change anything!\n";
+	List.Remove(4);
+	List.Remove(5);
+	List.Remove(6);
+	List.Remove(-1);
+	List.Remove(-10);
+	PrintList(List);
+
+	std::cout << "\nInserting Elements...!\n";
+	List.Insert("Is", 0);
+	List.Insert("?", 4);
+	List.Insert("Dumb", 2);
```

Comment:
JD showed us a change for this part of the test, please implement the modificaction
List.Insert("Dumb",3)